tasks/main.yml

```diff
@@ -1 +1,94 @@
-# role tasks
+# tasks file for azure-driver
+- import_role:
+    name: oasis_roles.system_repositories
+  vars:
+    system_repositories_configs:
+      - name: azure-cli
+        baseurl: https://packages.microsoft.com/yumrepos/azure-cli
+        gpgcheck: true
+        gpgkey: https://packages.microsoft.com/keys/microsoft.asc
+
+- name: Install Azure CLI
+  package:
+    name: azure-cli
+    state: present
+
+- name: Install jq
+  package:
+    name: jq
+    state: present
+
+- name: Ensure Azure credential folder exists
+  file:
+    path: "~/.azure/"
+    state: directory
+
+# These json files allow us to skip the az login interactive step.
+- name: Copy json files to ~/.azure
+  copy:
+    src: "{{ lookup('env', 'WORKSPACE') }}\
+/scenario/css_openshift/config/ansible/roles/azure_infra/files/{{ item }}"
+    dest: "~/.azure/{{ item }}"
+  with_items:
+    - azureProfile.json
+    - accessTokens.json
+
+# We can't guarantee the SP will be the same, so let's pull a new one every run
+# and update the credentials file
+- name: Get new service principal secret
+  command: "az ad sp create-for-rbac --role Contributor \
+            --name {{ azure_sp_name }}"
+  register: command_result
+  changed_when: command_result.rc == 0
+
+- name: Get Service Principal password
+  set_fact:
+    azure_sp_password: "{{ (command_result.stdout | from_json).password }}"
+
+- name: Get Service Principal Client ID
+  set_fact:
+    azure_sp_client_id: "{{ (command_result.stdout | from_json).appId }}"
+
+- name: Configure credentials file
+  template:
+    src: templates/azure_creds_file.j2
+    dest: ~/.azure/credentials
+    mode: '0755'
```

Comment:
Belt and suspenders applied. To tell yaml this is octal, you either need to add a leading zero to the int representation, or quote it as a string to let ansible handle the conversion. This does both. [The docs for template module](https://docs.ansible.com/ansible/latest/modules/template_module.html) make me think that the behavior when *both* rules are applied is undefined, so it's probably best to use one or the other. I personally prefer `mode: 0755`, but it doesn't much matter either way.